### PR TITLE
Feature/bbb 122 implement user roles in front end left hand menu

### PR DESCRIPTION
**Pull request by mgilgar**

### CODE REVIEW

File: acceptance-tests/src/test/java/uk/gov/service/bluebadge/test/acceptance/pages/site/AbstractSitePage.java [hunk 1]
```diff
@@ -38,6 +38,7 @@ public void openByPageName(final String pageName) {
 
   public void openByPageNameUnmapped(final String pageName) {
     String lookupUrl = TestContentUrls.lookupUrlUnmapped(pageName);
+    System.err.println("--------------lookupUrl: " + lookupUrl);
```
Comment (stephen-bealine) [inline acceptance-tests/src/test/java/uk/gov/service/bluebadge/test/acceptance/pages/site/AbstractSitePage.java]: Should be changed to an assert or removed

File: acceptance-tests/src/test/resources/features/site/badge-details.feature [hunk 2]
```diff
@@ -10,7 +10,7 @@ Feature: Dft BlueBadge LA View Badge Details
     When  I can click on the "Sign in" link
     When  I type username "abc@dft.gov.uk" and password "password"
     And   I can click Sign in button
-    Then I should see the page titled "New applications - GOV.UK Manage Blue Badges"
+    Then I should see the page titled "Manage users - GOV.UK Manage Blue Badges"
```
Comment (stephen-bealine) [inline acceptance-tests/src/test/resources/features/site/badge-details.feature]: Not sure about all of these changes to the first page. abc@def.gov.uk is a LA Admin. So they should be able to see the application page. Hence would have thought the landing page to remain the same

File: acceptance-tests/src/test/resources/features/site/sign_in_as_dft_admin.feature [hunk 3]
```diff
@@ -0,0 +1,29 @@
+@SignInDftAdmin @UsersRolesAndPermissionsScripts
```
Comment (stephen-bealine) [inline acceptance-tests/src/test/resources/features/site/sign_in_as_dft_admin.feature]: DFT User not really covered in this ticket. But no harm for the test. But will need to change, as the DFT user will not have a LA

File: acceptance-tests/src/test/resources/scripts/delete_users.sql [hunk 4]
```diff
@@ -0,0 +1,4 @@
+SET search_path TO usermanagement;
+
+-- Use an invalid postcode and name to delete test data
+DELETE FROM usermanagement.users WHERE id IN ('-1000', '-1001', '-1002', '-1003', '-1004', '-1005', '-1006', '-1007');
```
Comment (stephen-bealine) [inline acceptance-tests/src/test/resources/scripts/delete_users.sql]: Could just use a range.

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/config/SpringSecurityConfig.java [hunk 5]
```diff
@@ -68,8 +68,14 @@ protected void configure(HttpSecurity http) throws Exception {
         .authorizeRequests()
         .antMatchers("/sign-in", "/css/**", "/images/**", "/js/**", "/govuk/**")
         .permitAll()
+        .antMatchers("/new-applications", "/new-applications/**")
+        .hasAuthority(Permissions.FIND_APPLICATION.getPermissionName())
+        .antMatchers("/order-a-badge", "/order-a-badge")
```
Comment (stephen-bealine) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/config/SpringSecurityConfig.java]: duplicate ant macher

File: acceptance-tests/src/test/resources/features/site/badge-details.feature (hunk 2 above)
Comment (robsvaltech) [reply acceptance-tests/src/test/resources/features/site/badge-details.feature]: Confirmed with Dave that an anyone with FIND_USER role should be directed to Manage Users page

File: acceptance-tests/src/test/resources/features/site/sign_in_as_dft_admin.feature (hunk 3 above)
Comment (robsvaltech) [reply acceptance-tests/src/test/resources/features/site/sign_in_as_dft_admin.feature]: I think the DftAdmin user here has had the LA added as that is the current constraint for users.. We'll need to handle this in the ticket that implements dft admin users OR remove the test.

**Pull request merged**